### PR TITLE
LoadEventNexus duplicate code

### PR DESCRIPTION
**Pull request by peterfpeterson**
While helping out people loading data from LENS, I discovered that `LoadEventNexus` had an independent implementation of loading event monitors which gave different results from `LoadNexusMonitors`. Talking with developers at SNS, we decided to change the parameter names to clarify whether it was event or histogram monitors that were being loaded. The result is having `LoadNexusMonitors` always used from `LoadEventNexus` and documentation cleanup. The tests that were removed were verifying that broken behavior continued. There is much better coverage in `LoadNexusMonitors` already so the likelihood of new bugs being introduced should actually be lower.

**To test:**

Try loading your favorite nexus files with monitors and see if things still work. There were some CORELLI files that had to be tried with a particular (and strange) set of parameters in `LoadEventNexus`. Ticking the `LoadMonitors` option should be enough now.

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by martyngigg** [comment]
@peterfpeterson This has picked up a conflict.

File: Framework/DataHandling/src/LoadEventNexus.cpp [hunk 1]
```diff
@@ -210,27 +185,11 @@ void LoadEventNexus::init() {
...
-                  "will be loaded.");
-
-  declareProperty("LoadHistoMonitors", true,
-                  "Load histogram monitor in NeXus file both event monitor and "
-                  "histogram monitor found in NeXus file."
-                  "If both of LoadEventMonitor and LoadHistoMonitor are true, "
-                  "or both of them are false,"
-                  "then it is in the auto mode such that any existing monitor "
-                  "will be loaded.");
+  std::vector<std::string> options{"", "Events", "Histogram"};
```
Comment (martyngigg) [inline Framework/DataHandling/src/LoadEventNexus.cpp]: So this change will break user scripts - should we provide a deprecation warning in the Python layer?

Comment (peterfpeterson) [reply Framework/DataHandling/src/LoadEventNexus.cpp]: In common usage, people use the default which is "figure it out." It is only in the case of both histogram *and* events being in the file that the user has to intervene.

File: Framework/PythonInterface/mantid/simpleapi.py [hunk 2]
```diff
@@ -1075,6 +1075,17 @@ def algorithm_wrapper(*args, **kwargs):
...
             del kwargs["CoordinatesToUse"]
 
+        # a change in parameters should get a better error message
+        if algm.name() in ['LoadEventNexus', 'LoadNexusMonitors']:
+            for propname in ['MonitorsAsEvents', 'LoadEventMonitors', 'LoadHistoMonitors']:
+                if propname in kwargs:
+                    suggest = 'LoadOnly'
+                    if algm.name() == 'LoadEventNexus':
+                        suggest = 'MonitorsLoadOnly'
+                    msg = 'Deprecated property "{}" in {}. Use "{}" instead'.format(propname,
```
Comment (martyngigg) [inline Framework/PythonInterface/mantid/simpleapi.py]: :+1:

**Review by martyngigg — APPROVED** [review]
Refactoring improves the situation regarding loading the monitors code.

The clean up of the `LoadEventNexus` properties provides much better clarity over what happens when you load monitors with an event file. :+1:

**Pull request merged**